lib-model/lib-model-consume-cli/src/main/java/com/uber/nullaway/libmodel/LibModelInfoExtractor.java

```diff
@@ -0,0 +1,287 @@
+import com.github.javaparser.ast.body.FieldDeclaration;
+import com.github.javaparser.ast.body.InitializerDeclaration;
+import com.github.javaparser.ast.body.MethodDeclaration;
+import com.github.javaparser.ast.expr.AnnotationExpr;
+import com.github.javaparser.ast.expr.NormalAnnotationExpr;
+import com.github.javaparser.ast.type.ArrayType;
+import com.github.javaparser.ast.type.ClassOrInterfaceType;
+import com.github.javaparser.ast.type.TypeParameter;
+import com.github.javaparser.ast.visitor.VoidVisitorAdapter;
+import com.github.javaparser.utils.CollectionStrategy;
+import com.github.javaparser.utils.ParserCollectionStrategy;
+import com.github.javaparser.utils.ProjectRoot;
+import com.github.javaparser.utils.SourceRoot;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import java.io.DataOutputStream;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayDeque;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+public class LibModelInfoExtractor {
+
+  public static void main(String[] args) {
+    Map<String, MethodAnnotationsRecord> methodRecords = new LinkedHashMap<>();
+    LibModelInfoExtractor libModelInfoExtractor = new LibModelInfoExtractor();
+    // input directory
+    libModelInfoExtractor.processDirectory(args[0], methodRecords);
+    // output directory
+    libModelInfoExtractor.writeToAstubx(args[1], methodRecords);
+  }
+
+  public void processDirectory(String file, Map<String, MethodAnnotationsRecord> methodRecords) {
+    Path root = dirnameToPath(file);
+    AnnotationCollectorCallback mc = new AnnotationCollectorCallback(methodRecords);
+    CollectionStrategy strategy = new ParserCollectionStrategy();
+    // Required to include directories that contain a module-info.java, which don't parse by
+    // default.
+    strategy.getParserConfiguration().setLanguageLevel(LanguageLevel.JAVA_17);
+    ProjectRoot projectRoot = strategy.collect(root);
+
+    projectRoot
+        .getSourceRoots()
+        .forEach(
+            sourceRoot -> {
+              try {
+                sourceRoot.parse("", mc);
+              } catch (IOException e) {
+                System.err.println("IOException: " + e);
+              }
+            });
+  }
+
+  public void writeToAstubx(String outputPath, Map<String, MethodAnnotationsRecord> methodRecords) {
+    Map<String, String> importedAnnotations =
+        ImmutableMap.<String, String>builder()
+            .put("Nonnull", "javax.annotation.Nonnull")
+            .put("Nullable", "javax.annotation.Nullable")
+            .build();
+    DataOutputStream dos;
+    try {
+      Path opPath = Paths.get(outputPath);
+      Files.createDirectories(opPath.getParent());
+      dos = new DataOutputStream(Files.newOutputStream(opPath));
```

Comment:
Rather than explicitly calling `close()` later you should initialize `dos` in a try-with-resources block